Modules/Filtering/ImageSources/include/itkGenerateImageSource.hxx

```diff
@@ -63,6 +63,9 @@ GenerateImageSource<TOutputImage>::GenerateOutputInformation()
       outputPtr->SetSpacing(referenceImage->GetSpacing());
       outputPtr->SetOrigin(referenceImage->GetOrigin());
       outputPtr->SetDirection(referenceImage->GetDirection());
+      this->SetDirection(referenceImage->GetDirection());
+      this->SetStartIndex(referenceImage->GetLargestPossibleRegion().GetIndex());
+      this->SetSize(referenceImage->GetLargestPossibleRegion().GetSize());
```

Comment:
Calling this method will modify the m_Time of the filter. This may cause problems with executing the pipeline.

Perhaps the GridImageSource use of ["GetSize](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Filtering/ImageSources/include/itkGridImageSource.hxx#L56)" is incorrect? This would be tedious to check all derived classes are not accessing these parameters.

Alternatively, directly setting the Ivars, m_Direction, ..., would not modify the MTime.

Reply:
Good point. I'll set with the ivar.